### PR TITLE
DMP-2471: Create retention policy

### PR DESCRIPTION
**Pull request by deanbullock-hmcts**
### Jira link (if applicable)

[DMP-2471](https://tools.hmcts.net/jira/browse/DMP-2471)

### Change description ###

- Create retention policy screen and form component
- Add `POST /api/admin/retention-policy-types` endpoint to stub. Policy end date is generated by adding the duration to the start date time.
- When creating a retention policy, entering `NON_UNIQUE_POLICY_DISPLAY_NAME`, `NON_UNIQUE_POLICY_NAME` or `NON_UNIQUE_FIXED_POLICY_KEY` in their respective fields generates a 400 response from the stub simulating a bad request when creating a policy. These errors are handled by the component and the UI is updated accordingly.

### CODE REVIEW

File: src/app/admin/components/retention-policies/create-retention-policy/create-retention-policy.component.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component, OnInit, inject } from '@angular/core';
+import { Router } from '@angular/router';
+import { GovukHeadingComponent } from '@common/govuk-heading/govuk-heading.component';
+import { LoadingComponent } from '@common/loading/loading.component';
+import { ValidationErrorSummaryComponent } from '@common/validation-error-summary/validation-error-summary.component';
+import { ErrorSummaryEntry } from '@core-types/index';
+import { RetentionPoliciesService } from '@services/retention-policies/retention-policies.service';
+import { RetentionPolicyFormComponent } from '../retention-policy-form/retention-policy-form.component';
+import { HeaderService } from './../../../../core/services/header/header.service';
```
Comment (DavidTMann) [inline src/app/admin/components/retention-policies/create-retention-policy/create-retention-policy.component.ts:12]: @services alias here please

File: cypress/e2e/functional/admin-portal/retention-policies-spec.cy.js [hunk 2]
```diff
@@ -8,24 +8,32 @@ describe('Admin - Retention Policies screen', () => {
...
+    cy.get('#days').type('3');
+    cy.get('#startDate').type('10/10/3000');
+    cy.get('#start-time-hour-input').type('23');
+    cy.get('#start-time-minutes-input').type('59');
+
+    cy.contains('Save').click();
+
+    cy.get('app-govuk-banner').contains('Retention policy created');
+
+    cy.get('#active-policies-table').contains('New Policy');
```
Comment (DavidTMann) [inline cypress/e2e/functional/admin-portal/retention-policies-spec.cy.js:37]: Where a11y @ tho?

Comment (deanbullock-hmcts) [reply cypress/e2e/functional/admin-portal/retention-policies-spec.cy.js:41]: Nice one - fixed

File: src/app/admin/components/retention-policies/create-retention-policy/create-retention-policy.component.ts (hunk 1 above)
Comment (deanbullock-hmcts) [reply src/app/admin/components/retention-policies/create-retention-policy/create-retention-policy.component.ts]: Auto import done me dirty, fixed

**Review by DavidTMann — APPROVED** [review]

**Pull request merged**